video/out/aspect.c

```diff
@@ -155,6 +151,39 @@ void mp_get_src_dst_rects(struct mp_log *log, struct mp_vo_opts *opts,
 
     struct mp_rect dst = {0, 0, window_w, window_h};
     struct mp_rect src = {0, 0, src_w,    src_h};
+    if (mp_image_crop_valid(video))
+        src = video->crop;
+
+    if (vo_caps & VO_CAP_ROTATE90) {
```

Comment:
This logic is only needed if we have a crop active right? I think it would be better to add that condition here so it's clearer what this is actually for (ensuring the crop dimensions are correct when rotating).

Reply:
Not only, for crop, image dimensions are also swapped, there is also
```
MPSWAP(int, src_w, src_h);
MPSWAP(int, src_dw, src_dh);
```
We still need to rotate if not cropping, just then it is easy case, because we don't have offsets.

I was thinking how to make it shorter, but this is best I could do, factoring it to separate function also doesn't help much.

Reply:
Oh right the old code got moved. How do you feel about putting the old part back (`if (video->rotate % 180 == 90 && (vo_caps & VO_CAP_ROTATE90))`, taking out the two `MPSWAP` calls here, and then making this logic apply only in the crop case? I get that it's effectively a no-op (besides the swap part) if there's no crop but it's not exactly clear.

Reply:
Oh wait but that would make `struct mp_rect src = {0, 0, src_w,    src_h};` wrong wouldn't it for crops. Maybe just leave it as is....

Reply:
> Oh wait but that would make struct mp_rect src = {0, 0, src_w,    src_h}; wrong wouldn't it for crops. Maybe just leave it as is....

Yep, that's the point, need to rotate always, before we could shortcut with only swapping W/H, but now it doesn't flow with the code.  

I could factor it for `rotate_rect()` function, but then still need to swap W/H before, so it would duplicate the `video->rotate % 180 == 90` check. Do you think it would be better?

Reply:
Yeah let's go with that for clarity's sake.

Reply:
Ok, done.